File: practica2/c++/dup.cc
#include "dup.h"
#include <iostream>
#include <stack>


using namespace std;

void dup::ejecutar(stack<int32_t> &pila, int32_t& pc) const
{
    int32_t tmp1 = pila.top();
    pila.pop();
    pila.push(tmp1);
    pila.push(tmp1);
    pc++;
}
string dup::nombre() const
{
	return "dup";
}

File: practica2/java/Dup.java
import java.util.Stack;

/**
 * dup
 */
public class Dup extends Inst
{
    @Override
    public int ejecutar(Stack<Integer> pila, int pc){
    int tmp1 = pila.pop();
    pila.push(tmp1);
    pila.push(tmp1);
    pc++;
    return pc;
    }

    @Override
    public String nombre(){
        return "dup";
    }
}

